test/e2e/network/proxy.go

```diff
@@ -309,9 +310,10 @@ var _ = common.SIGDescribe("Proxy", func() {
 					RestartPolicy: v1.RestartPolicyNever,
 				}}, metav1.CreateOptions{})
 			framework.ExpectNoError(err, "failed to create pod")
-
+			framework.ExpectNoError(e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, "agnhost", f.Namespace.Name))
```

Comment:
I wonder if 
```
err = wait.PollImmediate(podRetryPeriod, podRetryTimeout, checkPodStatus(f, "test=response"))
```
is now a duplicate and we should consolidate them in e2epod.WaitForPodNameRunningInNamespac

Reply:
WaitForPodNameRunningInNamespace waits 5 min
https://github.com/pacoxu/kubernetes/blob/16adb0f7848d437b86a83b01650f45031f9fd861/test/e2e/framework/pod/wait.go#L52-L53

this has lower timeouts

https://github.com/kubernetes/kubernetes/blob/16adb0f7848d437b86a83b01650f45031f9fd861/test/e2e/network/proxy.go#L61-L62

Reply:
I change it to check the status after `WaitForPodNameRunningInNamespace`.